apps/studio/src/features/dashboard/components/DeleteResourceModal/DeleteResourceModal.tsx

```diff
@@ -0,0 +1,134 @@
+import { useState } from "react"
+import {
+  HStack,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from "@chakra-ui/react"
+import {
+  Button,
+  Checkbox,
+  ModalCloseButton,
+  useToast,
+} from "@opengovsg/design-system-react"
+import { ResourceType } from "~prisma/generated/generatedEnums"
+
+import { isAllowedToHaveChildren } from "~/utils/resources"
+import { trpc } from "~/utils/trpc"
+import { DeleteCollectionModalState } from "../../types"
+
+const getResourceLabel = (
+  resourceType: ResourceType,
+): Lowercase<Extract<ResourceType, "Page" | "Collection" | "Folder">> => {
+  if (!isAllowedToHaveChildren(resourceType)) {
+    return "page"
+  }
+
+  if (resourceType === ResourceType.Collection) {
+    return "collection"
+  }
+
+  return "folder"
+}
+
+const getWarningText = (resourceType: ResourceType): string => {
+  if (!isAllowedToHaveChildren) {
+    return "This cannot be undone"
+  }
+
+  if (resourceType === ResourceType.Collection) {
+    return "All pages under this collection will be deleted. This cannot be undone."
+  }
+
+  return "All folders and pages under this folder will be deleted. This cannot be undone."
+}
+
+interface DeleteResourceModalProps extends DeleteCollectionModalState {
+  onClose: () => void
+  siteId: number
+}
+
+export const DeleteResourceModal = (
+  props: DeleteResourceModalProps,
+): JSX.Element => {
+  return (
+    <Modal isOpen={!!props.isOpen} onClose={props.onClose}>
+      <ModalOverlay />
+      <DeleteResourceModalContent {...props} />
+    </Modal>
+  )
+}
+
+const DeleteResourceModalContent = ({
+  title,
+  resourceId,
+  resourceType,
+  siteId,
+  isOpen,
+  onClose,
+}: DeleteResourceModalProps) => {
+  const label = getResourceLabel(resourceType)
+  const utils = trpc.useUtils()
+  const toast = useToast()
+  const [isChecked, setIsChecked] = useState(false)
+  const { mutate, isLoading } = trpc.resource.delete.useMutation({
+    onSettled: onClose,
+    onSuccess: async () => {
+      // TODO: here and elsewhere, we should aim to simplify our query pattern
+      // such that the invalidation logic is clear
+      await utils.resource.list.invalidate()
+      await utils.resource.getChildrenOf.invalidate()
+      await utils.collection.list.invalidate()
+      toast({ title: "Folder created!", status: "success" })
+    },
+    onError: (err) => {
+      toast({
+        title: "Failed to create folder",
+        status: "error",
+        // TODO: check if this property is correct
+        description: err.message,
```

Comment:
does delete procedure return message as a property? are we throwing a trpc error?

Reply:
we're throwing an error - i hvaen't validated this but the `err` has a `message` property, which i assumed (unvalidated) that it's the one from `trpcerror`